File: bindings/java/examples/Prompt02Reports.java
package io.wellfriendpdf.examples;

import java.nio.file.Files;
import java.nio.file.Path;
import io.wellfriendpdf.Wellfriend;

public final class Prompt02Reports {
    private Prompt02Reports() {
    }

    public static void main(String[] args) throws Exception {
        if (args.length != 1) {
            System.err.println("usage: Prompt02Reports <pdf>");
            System.exit(2);
        }

        Path input = Path.of(args[0]);
        try (WellfriendPdf.Document doc = WellfriendPdf.Document.open(input)) {
            System.out.println(doc.securityReportJson());
            WellfriendPdf.BinaryResult sanitized = doc.sanitize("balanced");
            Files.write(input.resolveSibling(input.getFileName() + ".sanitized.pdf"), sanitized.bytes());
            System.out.println(sanitized.reportJson());
        }
    }
}

File: bindings/dotnet/examples/Prompt02Reports.cs
using WellfriendPdf;

if (args.Length != 1)
{
    Console.Error.WriteLine("usage: Prompt02Reports <pdf>");
    return 2;
}

using var doc = WellfriendDocument.Open(args[0]);
Console.WriteLine(doc.SecurityReportJson());

var sanitized = doc.Sanitize("balanced");
var outPath = Path.ChangeExtension(args[0], ".sanitized.pdf");
File.WriteAllBytes(outPath, sanitized.Bytes);
Console.WriteLine(sanitized.ReportJson);
return 0;
